tests/integration/test_fim/test_registry/test_nodiff/test_no_diff.py

```diff
@@ -0,0 +1,161 @@
+
+
+    def no_diff_validator(event):
+        """Validate content_changes value is truncated if the file is set to no_diff"""
+        if truncated:
+            assert '<Diff truncated because nodiff option>' in event['data'].get('content_changes'), \
+                f'content_changes is not truncated'
+        else:
+            assert '<Diff truncated because nodiff option>' not in event['data'].get('content_changes'), \
+                f'content_changes is truncated'
+
+
+    registry_value_cud(key, subkey, wazuh_log_monitor, arch=arch, value_list=values,
+                     time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled',
+                     min_timeout=global_parameters.default_timeout, triggers_event=True,
+                     validators_after_update=[report_changes_validator, no_diff_validator])
+
+
+@pytest.mark.parametrize('key, subkey, arch, value_name, content, truncated, tags_to_apply', [
+    (key, sub_key_1, KEY_WOW64_64KEY, no_diff_value, "test_content", True, {'no_diff_regex'}),
+    (key, sub_key_1, KEY_WOW64_64KEY, "some_value", "test_content", False, {'no_diff_regex'}),
+    (key, sub_key_1, KEY_WOW64_32KEY, no_diff_value, "test_content", True, {'no_diff_regex'}),
+    (key, sub_key_1, KEY_WOW64_32KEY, "some_value", "test_content", False, {'no_diff_regex'}),
+    (key, sub_key_2, KEY_WOW64_64KEY, no_diff_value, "test_content", True, {'no_diff_regex'}),
+    (key, sub_key_2, KEY_WOW64_64KEY, "some_value", "test_content", False, {'no_diff_regex'})
+])
+def test_no_diff_regex(key, subkey, arch, value_name, content, truncated, tags_to_apply,
+                  get_configuration, configure_environment, restart_syscheckd,
+                  wait_for_fim_start):
+    """
+    Check the only files detected are those matching the restrict regex
+
+    Parameters
+    ----------
+    key : str
+        Root key (HKEY_*)
+    subkey : str
+        path of the registry.
+    arch : str
+        Architecture of the registry.
+    value_name : str
+        Name of the value that will be created
+    truncated : bool
+        True if an event must be generated, False otherwise.
+    tags_to_apply : set
+        Run test if match with a configuration identifier, skip otherwise.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+    values = {value_name: content}
+
+    def report_changes_validator(event):
+        """Validate content_changes attribute exists in the event"""
+        for value in values:
+            folder_str = "{} {}".format("[x32]" if arch == KEY_WOW64_32KEY else "[x64]", sha1(os.path.join(key, subkey).encode()).hexdigest())
+            diff_file = os.path.join(WAZUH_PATH, 'queue', 'diff', 'registry', folder_str, sha1(value.encode()).hexdigest())
+            assert os.path.exists(diff_file), f'{diff_file} does not exist'
+            assert event['data'].get('content_changes') is not None, f'content_changes is empty'
+
+
+    def no_diff_validator(event):
```

Comment:
There are too many blank line before this method declaration.
```suggestion
    def no_diff_validator(event):
```